specutils/analysis/template_match.py

```diff
@@ -0,0 +1,36 @@
+from ..spectra.spectrum1d import Spectrum1D
+from ..spectra.spectrum_collection import SpectrumCollection
+from ..manipulation import FluxConservingResample
+from scipy.stats import chisquare
+
+def _template_match(observed_spectrum, template_spectrum):
+    # resample the template spectrum to match the wavelength of the observed spectrum
+    fluxc_resample = FluxConservingResample()
+    output_spectrum1D = fluxc_resample(template_spectrum, observed_spectrum.wavelength)
```

Comment:
I would suggest change `output_spectrum1D` to `template_spectrum1D` just so it is clear that is the resampled template.